File: ruby/rna-transcription/rna_transcription.rb
class Complement
  Complement::VERSION = 3

  def self.of_dna(dna)
    strand_validation(dna)
    rna = ""
    dna.chars.map do |char|
      rna << char.tr('GCTA', 'CGAU')
    end
      rna
  end

  def self.strand_validation(dna)
    validation = dna.gsub(/[GCTA]/, "")
    raise ArgumentError if !validation.empty?
  end

end

File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {
  this.toRna = function (dna) {
    var dnaMapping = { G: 'C',
                       C:'G',
                       T: 'A',
                       A: 'U' };
    var rna = '';
    for (n = 0; n < dna.length; n++) {
      rna += dnaMapping[dna[n]]
      // console.log(dna);
    };
    return rna;
  };
};
module.exports = DnaTranscriber
